tests/Fixer/Operator/StandardizeIncrementFixerTest.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\Operator;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Operator\StandardizeIncrementFixer
+ */
+final class StandardizeIncrementFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideFixCases
+     */
+    public function testFix($expected, $input = null)
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases()
+    {
+        return [
+            [
+                '<?php ++$i;',
+                '<?php $i = $i + 1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i=$i+1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i   =  $i  +   1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i += 1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i+=1;',
+            ],
+            [
+                '<?php --$i;',
+                '<?php $i = $i - 1;',
+            ],
+            [
+                '<?php --$i;',
+                '<?php $i=$i-1;',
+            ],
+            [
+                '<?php --$i;',
+                '<?php $i -= 1;',
+            ],
+            [
+                '<?php --$i;',
+                '<?php $i-=1;',
+            ],
+            [
+                '<?php ++$i;',
+                '<?php $i /* foo */ = $i /* bar */ + 1;',
```

Comment:
Is this the desired behaviour?

Reply:
please don't throw away comments, no matter how badly placed, leave them somewhere near their original place.

Reply:
any news on that ?